controllers/reconcilers/configuration/prometheus.go

```diff
@@ -122,22 +122,42 @@ func (r *Reconciler) createAdditionalScrapeConfigSecret(cr *v1.Observability, ct
 
 func (r *Reconciler) getOpenshiftMonitoringCredentials(ctx context.Context) (string, string, error) {
 	secret := &kv1.Secret{}
+	osVersionToCompareString := "4.10.0"
+	var isDatasourceV2 bool
+
+	currentOSVersionString, err := utils.GetClusterOSVersion(ctx, r.client)
+	if err != nil {
+		return "", "", err
+	}
+
+	isDatasourceV2, err = utils.HasNewerOrSameClusterVersion(currentOSVersionString, osVersionToCompareString)
```

Comment:
Could we leave a comment about how the cluster version relates to the version of grafanaDatasource

Reply:
Sure. I'll add that comment